### PR TITLE
Unhandled error incident disappears

### LINKED ISSUE #7160: Resolving an uncaught error incident does not lead to new incident
**Describe the bug**

Currently, unhandled error event incidents cannot be resolved (see #6000). Instead, the expected behavior is just to resolve the incident and create a new one. However, when resolving an unhandled error event incident associated to a job, the resolve command is both rejected and simultaneously accepted.

Relates to https://github.com/camunda-cloud/zeebe/issues/7127 (similar)

**To Reproduce**

- create a job
- throw an error for the job and make sure it's not caught by the process
- resolve the incident
- notice that incident:resolve command was accepted
- notice that incident:resolved event was written to the log
- notice that incident:resolve rejection was written to the log
- notice that no incident exists for the failure any more and that the processing does not continue

**Expected behavior**

The incident:resolve is accepted, incident:resolved is written to the log and a new incident:created is written to the log

**Environment:**
 - Zeebe Version: 1.0.0

### PR DESCRIPTION
**Pull request by korthout**
## Description

Adds a test to reproduce the bug described in #7160 and provides a simple fix: after resolving the incident, create a new incident with the same record value.

## Related issues

closes #7160

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Tests for incidents created due to UNHANDLED_ERROR_EVENT associated to jobs */
+public final class JobUnhandledErrorEventIncidentTest {
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java:26]: (optional) We could merge this test with the related tests in `io.camunda.zeebe.engine.processing.incident.ErrorEventIncidentTest`.

**Review by saig0 — APPROVED** [review]
Nice :+1:  

Please have a look at my comment before merging :cake:

File: engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java (hunk 1 above)
Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java:26]: Interesting, I did not consider this to be about the same topic. But it is. I'll move it there.

Side note, just now discovered that one of the tests in ErrorEventIncidentTest takes 10 seconds. I might have a look at that as well

Comment (saig0) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java:26]: Sounds good :+1:

Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java]: @saig0 I've cleaned-up and sped-up that test class a bit, and then moved the new test there. Please have another look. Only changed commits are the first 3 (i.e. with type `test`).

**Review by saig0 — APPROVED** [review]
Looks good :green_apple:

**Comment by korthout** [comment]
bors merge

**Pull request merged**